### ISSUE TITLE
Binder for read-only components (Label..)

### ISSUE THREAD

**Issue by darkwitch** [vaadin/framework#8540]
It would be great if Binder could be used to work with read-only components, such as Label. 

It's a very common use case where you want to display some 'properties' along with editable fields, for ex: profile edit form:

- username / last login are read-only, 
- first name, last name, password are editable

or have an auto-refreshing view of some values. 

Most binding frameworks support this.

Currently using Binder we can only Bind **HasValue<?>** Components, so we have to set label values manually. That's a big pain.

I did a quick test and binding a custom BindableLabel as below works fine, but it will not be Designer friendly, as it doesn't support custom components (they are displayed as Containers). 

```
    public class BindableLabel extends Label implements HasValue<String> {

        @Override
        public Registration addValueChangeListener(ValueChangeListener<String> listener) {
            return () -> {}; //not supported for Label, might consider throwing UnsupportedOperationException
        }

        @Override
        public void setRequiredIndicatorVisible(boolean requiredIndicatorVisible) {
            //not supported for Label, might consider throwing UnsupportedOperationException
        }

        @Override
        public boolean isRequiredIndicatorVisible() {
            return false;
        }

        @Override
        public void setReadOnly(boolean readOnly) {
            //not supported for Label, might consider throwing UnsupportedOperationException
        }

        @Override
        public boolean isReadOnly() {
            return true;
        }
    }
```

I do not know the whole scope of potential problems with this approach, bu the simplest solution could be to implement HasValue directly in Label....

This is a followup of a discussion about this [here](https://vaadin.com/forum#!/thread/3376266).

**Comment by jdahlstrom** [comment]
As far as I know, a read-only TextField should work fine in Vaadin 8 and is a semantically correct solution.

**Comment by elmot** [comment]
[A code example with three different approaches](https://github.com/elmot/tipsandtricksbinder/blob/master/src/main/java/org/vaadin/tipsandtricks/TipsAndTricksUI.java)

- Read-only TextField
- Label subclassing
- Generic HasValue implementation

Pick whatever you like more

**Comment by Legioth** [comment]
The best would still be if I could just do something like this:
```
binder.bind(myLabel, Item::getId);

binder.bind(myProgressBar).withConverter(value -> value / 100).bind(Item::getProgress);
```

**Comment by envas** [comment]
TextField is semantically correct but does not replace some nice Label features like content type HTML. In Vaadin 7 we are using, for example, bound FontAwesome labels in Dashboards.

**Comment by rmuller** [comment]
A common use-case is binding beans with read/write properties where some properties are read only **in the UX**. In this case it would be great to have something like:

````
binder.forField(hasValue)
    .asReadOnly().   // new, consistent with asRequired()
    .bind(beanPropertyName);
````

Code example to make things clear: https://gist.github.com/rmuller/dcb6b6db1d06db185f536db477419fb2

**Comment by sebastianfilke** [comment]
This would be an awesome feature. Any plans for a release?

**Comment by maticpetek** [comment]
Hi, binding Labels would really be nice features. Any plans / workround for this?

**Comment by tsuoanttila** [comment]
There's a `ReadOnlyHasValue` in a [repo](https://github.com/elmot/tipsandtricksbinder/blob/master/src/main/java/org/vaadin/tipsandtricks/binder/ReadonlyHasValue.java) from @elmot. That should allow using practically any component as a HasValue for binder purposes.

**Comment by bsutton** [comment]
> binder.bind(myProgressBar)
> .withConverter(value -> value / 100)
> .bind(Item::getProgress);

I would agree @Legioth suggested method as being the most elegant and consistent.

**Comment by elmot** [comment]
Fixed with #10643
